kikit/export.py

```diff
@@ -1,21 +1,41 @@
 # Based on https://github.com/KiCad/kicad-source-mirror/blob/master/demos/python_scripts_examples/gen_gerber_and_drill_files_board.py
 import sys
 import os
+from enum import Enum
+
 from pcbnewTransition import pcbnew
 from pcbnew import *
 
+
+class LayerToPlot(Enum):
+    CuTop = (pcbnew.F_Cu, "Top layer")
+    CuBottom = (pcbnew.B_Cu, "Bottom layer")
+    PasteTop = (pcbnew.F_Paste, "Paste top")
+    PasteBottom = (pcbnew.B_Paste, "Paste bottom")
+    SilkTop = (pcbnew.F_SilkS, "Silk top")
+    SilkBottom = (pcbnew.B_SilkS, "Silk bottom")
+    MaskTop = (pcbnew.F_Mask, "Mask top")
+    MaskBottom = (pcbnew.B_Mask, "Mask bottom")
+    EdgeCuts = (pcbnew.Edge_Cuts, "Edges")
+    CmtUser = (pcbnew.Cmts_User, "V-CUT")
+    AdhesiveTop = (pcbnew.F_Adhes, "Adhesive top")
+    AdhesiveBottom = (pcbnew.B_Adhes, "Adhesive bottom")
+
+    def __init__(self, id: int, description: str):
+        self.id = id
+        self.description = description
+
```

Comment:
I am unsure about this approach. I understand the need to provide "commonly used values"; however, this seems like an abuse of Enum a breaks type annotations.

If we want to allow the users to provide custom descriptions to the layers (as it was possible so far), this approach breaks the type system. The users will be forced to pass new tuples that, however, aren't part of the enum.

If we want to forbid it, I find a much nicer solution not to introduce new types and instead, internally have a function that maps layers to descriptions.

I understand the need for giving some type safety here - the code really needs it. But in that case, wouldn't a named tuple or a dataclass be a much better solution? Something like:
```
@dataclass
class LayerToPlot:
	layer: Layer
	description: str
```

And then, if required, provide some default values.

Or is there something I miss about using the enum?

Reply:
IIRC I changed this because the cutout layer name was wrong. I didn't consider the custom descriptions. Dataclass looks like a good approach, I'll give that a try